extensions/exchanges/gdax/exchange.js

```diff
@@ -1,20 +1,53 @@
 var Gdax = require('gdax')
-  , path = require('path')
-  , colors = require('colors')
-  , numbro = require('numbro')
 
 module.exports = function container (get, set, clear) {
   var c = get('conf')
 
-  var public_client = {}, authed_client
+  var public_client = {}, authed_client, websocket_client = {}, websocket_cache = {}
 
   function publicClient (product_id) {
     if (!public_client[product_id]) {
+      websocketClient(product_id)
       public_client[product_id] = new Gdax.PublicClient(product_id, c.gdax.apiURI)
```

Comment:
To use new API we can change: Gdax.PublicClient(product_id, c.gdax.apiURI) to Gdax.PublicClient(c.gdax.apiURI)

Reply:
I think I'll probably just pin the dependency for now (via PR from @fuzzyTew above). The idea of going to the new API does appeal to me, but as we just experienced, it's much better to have a consistent working install than to have the dependency update underneath us and potentially break things for people who might not have the expertise to diagnose such issues.

Reply:
yeah it's a good change to set specific API version so stuff doesn't break uncontrolled.